### DISCUSSION TITLE
FFmpeg 解析使用体验反馈

### DISCUSSION THREAD

**Discussion by Richasy** [Richasy/Bili.Uwp#1322] [Polls]
2.2207.2.0 中提供了 FFmpeg 的解析选择。你可以在 `设置` -> `播放器控制` -> `播放器类型` 中切换。在切换为 FFmpeg 后，你还可以进一步选择使用软解还是硬解。

**Comment by fxzxmicah** [comment]
内啥，挺奇怪的，我没切换FFmpeg，但是好像更新后直播切换所有清晰度都不会遇到只有声音没画面的情况了。

**Reply by Richasy** [reply]
除了引入ffmpeg，也同时进行了其它的逻辑调整，可能顺便修复了你遇到的问题吧😂

**Reply by fxzxmicah** [reply]
哦，其实没有100%解决，但是出问题的几率貌似低了不少。搭配FFmpeg就解决100%问题了，而且实测硬件解码更好。

**Comment by St-Cai** [comment]
这个版本不上架微软应用商店吗

**Reply by St-Cai** [reply]
FFmpeg - 自动 - 切换全屏黑屏
FFmpeg - 软件解码 - 切换全屏黑屏
FFmpeg - 硬件解码 - 切换全屏正常

感觉目前切换全屏的体验很差，表现为闪屏、卡顿、动画异常等，而且时不时会出现白屏/黑屏仅声音的现象

**Reply by St-Cai** [reply]
相比之下，xiaoyaocz的聚合直播软件[AllLive](https://github.com/xiaoyaocz/AllLive)的体验我觉得更好。

**Reply by hujianglang** [reply]
换个思路，调整布局，达成全屏效果，是不是好点

**Comment by CHminggao** [comment]
使用ffmpeg会莫名其妙的跳过某些视频
`BV17N4y1373p`
像这个视频里面有个`火影疾风传OP3 青鸟` 8s左右就会自动切到下一个视频，原生就不会

**Comment by returnL** [comment]
用了FFmpeg之后，视频会放着放着突然跳到结尾了

**Reply by returnL** [reply]
出现在全屏播放的时候

**Reply by Alwaysbesideyou** [reply]
同样遇到

**Reply by Alex-syz** [reply]
同样遇到

**Comment by fxzxmicah** [comment]
使用FFmpeg时发现一个问题，就是在播放途中，未进行任何操作，视频声音就消失了，退出播放重新进入以后恢复正常。
以上突然跳到视频结尾的问题我这里也存在。

**Comment by muxueBlue** [comment]
昨天更新了一下，下载废了老鼻子劲了。安装完新版，转手就把原来的版本卸载了。我后悔了，比原来的卡。不过没关系。加油

**Comment by SA-Root** [comment]
这次好像解决了之前我机器上核显不能看高码率直播的问题，并且全窗口模式的GPU占用似乎改善了很多。不过部分直播间只有声音无画面的问题还是存在，无论使用核显还是独显，原生还是ffmpeg均无法解决问题。

**Comment by fxzxmicah** [comment]
其实就是解决了现有bug并引入新的bug。😂

**Comment by fxzxmicah** [comment]
@Richasy 
在Windows11 22H2 build 22622（22621应该相同）中，微软应该是优化了原生视频解码，实测在没有更新显卡驱动的条件下使用**核显**解码相较于build 22000，原生解码体验已经优于FFmpeg，反而FFmpeg在新版本系统中闪退严重，几乎到了不可用的程度（所以我是否可以提议移除FFmpeg以减轻开发负担呢，提议罢了，交由开发者定夺）。
另外，在build 22000中频出问题的高画质（原生解码模式下），在22622中应该是得到了修复，我不敢说每个主播都没问题了，但是我随机测试了几位主播，并切换使用了所有画质，问题复现率0%，我想这个旷日持久，自项目起始就存在的问题应该可以说是终于得到了解决吧。
需要提出的一点是，在测试过程中，出现了可能是网络或者其它什么问题导致的直播卡住的情况，这时使用刷新会使整个软件卡死，而在正常播放时点击刷新则不会出现此问题（在日常使用中也感觉到了直播卡住时使用刷新导致卡死的概率很高，但是不清楚是不是100%会发生）。
